### PR TITLE
Sumcoin support for electrumx

### PR DESCRIPTION
**Pull request by sumcoinlabs**
coins.py and tests completed for Sumcoin

### CODE REVIEW

**Comment by kyuupichan** [comment]
There are 2 errors - please fix.  For pycodestyle you need 2 blank lines around class definitions

**Comment by 5053s** [comment]
> There are 2 errors - please fix. For pycodestyle you need 2 blank lines around class definitions

Can you be a little more specific about the issue please?  What file is this problem in?   Also please advise the method used to retrieve the "blocks" data from the daemon for use in tests/blocks/coin.json

image

**Comment by sumcoinlabs** [comment]
> There are 2 errors - please fix. For pycodestyle you need 2 blank lines around class definitions

Please review and advise what's wrong

File: tests/blocks/sumcoin_mainnet_837000.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "2350b84170ea4552a83f8df2add1391f4769a25a6dba34eac3887377f873f8d2"
+  ],
+  "time": 1579035712,
+  "mediantime": 1579035458,
+  "nonce": 2166465929,
+  "bits": "1b6574a3",
+  "difficulty": 645.9475093604372,
+  "chainwork": "00000000000000000000000000000000000000000000000011387b4a5d745891",
+  "previousblockhash": "1d22a9fa69ee10848f884077567021423bb72a7d779f1d70e0339e9981d84f15",
+  "nextblockhash": "134b64c0ea14027395832279150f6d35b68453df3dcc1fa59f8692b0ec3f298a"
```
Comment (SomberNight) [inline tests/blocks/sumcoin_mainnet_837000.json]: missing `,`

File: electrumx/lib/coins.py [hunk 2]
```diff
@@ -874,6 +874,24 @@ class BitcoinNolnet(BitcoinCash):
     RPC_PORT = 28332
     PEER_DEFAULT_PORTS = {'t': '52001', 's': '52002'}
 
+class Sumcoin(Coin):
```
Comment (SomberNight) [inline electrumx/lib/coins.py]: E302 expected 2 blank lines, found 1

File: electrumx/lib/coins.py [hunk 3]
```diff
@@ -874,6 +874,24 @@ class BitcoinNolnet(BitcoinCash):
...
+	WIF_BYTE = bytes.fromhex("bf")
+	GENESIS_HASH = ('37d4696c5072cd012f3b7c651e5ce56a'
+					'1383577e4edacc2d289ec9b25eebfd5e')
+	DESERIALIZER = lib_tx.DeserializerSegWit
+	TX_COUNT = 976394
+	TX_COUNT_HEIGHT = 659520
+	TX_PER_BLOCK = 2
+	RPC_PORT = 3332
+	REORG_LIMIT = 800
+	PEERS = []
```
Comment (SomberNight) [inline electrumx/lib/coins.py]: re the whole block, use spaces not tabs:
E101 indentation contains mixed spaces and tabs

**Comment by SomberNight** [comment]
> Please review and advise what's wrong

Read the Travis job log re the errors. (did it for you now)

File: electrumx/lib/coins.py (hunk 3 above)
Comment (jakesum) [reply electrumx/lib/coins.py]: updated spacing between classes.  What indentation issues are you seeing exactly?

**Comment by kyuupichan** [comment]
Your test and code style are still failing; please see the report under Details above.

**Comment by sumcoinlabs** [comment]
> Your test and code style are still failing; please see the report under Details above.

The methods have been copied as identically as we can tell to other coins.

File: electrumx/lib/coins.py (hunk 2 above)
Comment (sumcoinlabs) [reply electrumx/lib/coins.py]: Changes have been made but still fails to build.  Suggestion please

**Comment by kyuupichan** [comment]
Have you read the logs undef the Details button where it says failed?  https://travis-ci.org/kyuupichan/electrumx/jobs/640809765?utm_medium=notification&utm_source=github_status

**Comment by sumcoinlabs** [comment]
> https://travis-ci.org/kyuupichan/electrumx/jobs/640809765?utm_medium=notification&utm_source=github_status

Yes we have.  It seems to come down to a spacing issue but we have not been able to figure out what or where the issue exists.  Can you assist?

**Comment by kyuupichan** [comment]
"electrumx/lib/coins.py:878:50: W291 trailing whitespace" suggests you have spaces or tabs at the end of line 878

**Comment by kyuupichan** [comment]
Your own test for Sumcoin seems to be failing

**Comment by sumcoinlabs** [comment]
> Your own test for Sumcoin seems to be failing

The spacing was removed.  Not sure what is causing this to have so much trouble

**Comment by sumcoinlabs** [comment]
> Your own test for Sumcoin seems to be failing

Removed Block info in tests.   What else could we be missing?

=================================== FAILURES ===================================
_________________________ test_block[block_details86] __________________________
block_details = (<class 'electrumx.lib.coins.Sumcoin'>, {'bits': '1b6574a3', 'block': '01000000000101000000000000000000000000000000000...'chainwork': '00000000000000000000000000000000000000000000000011387b4a5d745891', 'difficulty': 645.9475093604372, ...})
    def test_block(block_details):
        coin, block_info = block_details
    
        raw_block = unhexlify(block_info['block'])
        block = coin.block(raw_block, block_info['height'])
    
>       assert coin.header_hash(
            block.header) == hex_str_to_hash(block_info['hash'])
E       AssertionError: assert b'\x99\xe7{\t...u\x14\xb2\x1a' == b'\xfb&Ei\x80...\x10\x85Q\xb3'
E         At index 0 diff: b'\x99' != b'\xfb'
E         Use -v to get the full diff
tests/test_blocks.py:65: AssertionError
=============================== warnings summary ===============================
tests/test_blocks.py::test_block[block_details1]

**Comment by jakesum** [comment]
> Your own test for Sumcoin seems to be failing

Thank you for your time on this.   Sumcoin is Passing all Tests now.

**Pull request merged**